File: p025.py
def fib():
    term, i, j = 0, 0, 1
    while True:
        yield term, i
        term += 1
        i = j + i
        j = i - j

for term, i in fib():
    if len(str(i)) >= 1000:
        print term
        break

File: go/src/p025/p025.go
package main

import (
    "fmt"
    "math/big"
)

func main() {
    term := 1
    for i, j := big.NewInt(1), big.NewInt(1); len(i.String()) < 1000; term++ {
        j.Add(j, i)
        i.Sub(j, i)
    }
    fmt.Println(term)
}
